plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/CloudSdkPreferences.java

```diff
@@ -57,7 +62,42 @@ static IEclipsePreferences getPreferenceNode() {
 
   @Override
   public void initializeDefaultPreferences() {
-    getPreferenceStore().setDefault(CLOUD_SDK_MANAGEMENT,
-        CloudSdkManagementOption.AUTOMATIC.name());
+    initializeDefaultPreferences(getPreferenceStore());
+  }
+
+  @VisibleForTesting
+  void initializeDefaultPreferences(IPreferenceStore preferences) {
+    if (CloudSdkManager.isManagedSdkFeatureEnabled()) {
+      if (!preferences.contains(CLOUD_SDK_MANAGEMENT)) {
+        // If the CLOUD_SDK_MANAGEMENT preference has not been set, then determine the
+        // appropriate setting. Note that CloudSdkPreferenceResolver only checks for
+        // the Managed Cloud SDK when CLOUD_SDK_MANAGEMENT has been explicitly set
+        // (i.e., this code has been run).
+        configureManagementPreferences(preferences, isCloudSdkAvailable());
+      }
+      preferences.setDefault(CLOUD_SDK_MANAGEMENT, CloudSdkManagementOption.AUTOMATIC.name());
+    }
+  }
+
+  /** Configure the managed SDK settings given current settings. */
+  @VisibleForTesting
+  static void configureManagementPreferences(
+      IPreferenceStore preferences, boolean cloudSdkAvailable) {
+    // has the user previously set the Cloud SDK path? has it been found in a well-known location?
+    if (!Strings.isNullOrEmpty(preferences.getString(CLOUD_SDK_PATH)) || cloudSdkAvailable) {
+      preferences.setValue(CLOUD_SDK_MANAGEMENT, CloudSdkManagementOption.MANUAL.name());
+    } else {
+      preferences.setValue(CLOUD_SDK_MANAGEMENT, CloudSdkManagementOption.AUTOMATIC.name());
+    }
+  }
+
+  /** Return {@code true} if the Cloud SDK is available. */
+  private static boolean isCloudSdkAvailable() {
+    try {
+      new CloudSdk.Builder().build();
```

Comment:
This is weird. We create a cloud sdk to find out if we can? There's no simpler way? And this isn't a method in appengine-plugins-core?

Reply:
It's not a trivial change in `CloudSdk`.  Basically the `CloudSdk.Builder` consults a set of resolvers, found using Java's `ServiceLoader`, to find the Cloud SDK location.